apps/src/applab/dontMarshalApi.js

```diff
@@ -1,9 +1,44 @@
+var errorHandler = require('./errorHandler');
+var outputApplabConsole = errorHandler.outputApplabConsole;
+var outputError = errorHandler.outputError;
+var ErrorLevel = errorHandler.ErrorLevel;
+
+var OPTIONAL = true;
+
+
+function outputWarning(errorString) {
+  var line = 1 + window.Applab.JSInterpreter.getNearestUserCodeLine();
+  outputError(errorString, ErrorLevel.WARNING, line);
+}
 
 // APIs designed specifically to run on interpreter data structures without marshalling
 // (valuable for performance or to support in/out parameters)
 //
 // dropletConfig for each of these APIs should be marked with dontMarshal:true
 
+
+function dmapiValidateType(funcName, varName, varValue, expectedType, opt) {
+  var properType;
+  if (typeof varValue !== 'undefined') {
+    if (expectedType === 'number') {
+      properType = (typeof varValue.data === 'number' ||
+                    (typeof varValue.data === 'string' && !isNaN(varValue.data)));
```

Comment:
OK, I see that I misread this -- you are accepting numerical strings when expectedType is 'number'. Makes sense.